File: solutions_11-20/Euler_15.py
## find all paths in 20 X 20 grid ##

import math
import timeit

start = timeit.default_timer()

new = math.factorial(40)

print "Answer: %s" % str(new/(math.factorial(20)*math.factorial(20)))
stop = timeit.default_timer()
print "Time: %s" % str(stop - start)

File: perl_solutions_11-20/Euler_15.pl
use strict;
use warnings;
use Time::HiRes qw( clock );
use Math::BigInt;


sub all_paths
{
    my $xstr = "20";
    my $xystr = "40";
    my $xy = Math::BigInt->new($xystr);
    my $x = Math::BigInt->new($xstr);
    $x->bfac();
    $xy->bfac();
    $x->bmul($x);
    return $xy->bdiv($x);
}

my $start = clock();
my $answer = all_paths();
my $stop = clock();
print "Answer: $answer\n";
printf "Time; %.5f\n", $stop - $start;
